### PR TITLE
new variable mungebit

### PR DESCRIPTION
**Pull request by peterhurford**
Wrapper around `multi_column_transformation` to make new variables.

Usage:

```R
list("Make new variable" = list(
  new_variable,
  function(Sepal.Width, Sepal.Length) Sepal.Width*2 + Sepal.Length*2,
  'Sepal.Perimeter'
))
```

Note that unlike `multi_column_transformation`, the input arguments are automatically inferred from the function passed in.  However, you can still specify inputs as a fourth argument if you wish:

 ```R
list("Make new variable" = list(
  new_variable,
  function(x, y) x*2 + y*2,
  'Sepal.Perimeter',
  c('Sepal.Width', 'Sepal.Length')
))
```

-

Also there's a wrapper around `new_variable` for replacing a variable, called `replace_variable`.  (Yes, this is a wrapper around a wrapper.  That's DRY code.)

Usage:
```R
list("Replace Sepal.Length with Sepal.Length + 1" = list(
  replace_variable,
  function(Sepal.Length) Sepal.Length + 1,
))
```

If you want to replace a variable using a function that takes more than one parameter, just put the variable you're replacing first in the function.  For example, this will replace `Sepal.Length`:

```R
list("Replace Sepal.Length with Sepal.Width + 1" = list(
  replace_variable,
  function(Sepal.Length, Sepal.Width) Sepal.Width + 1,
))
```

-

Also, to add a new variable without using previous variables, pass along the vector without needing a function:

```R
list("Add something from s3store" = list(
  add_variable,
  s3read('vector'),
  's3stored_data'
))
```

### CODE REVIEW

**Comment by peterhurford** [comment]
@robertzk

**Comment by peterhurford** [comment]
Note: added `replace_variable`, see updated PR description.

**Comment by peterhurford** [comment]
@iveksl2
> Tests?
Tests forthcoming

-
@iveksl2 
> Have you ever thought about using Dplyr along with magrittr using mutate for readability
Yes, that was my first thought, but dplyr doesn't play nicely with the train / predict differences.  Also, it's nice to have one less package dependency.

-

@FeiYeYe
> "so many changes @roxygen2"
Yes.  Everyone should standardize on the latest Roxygen: `install_github('klutometis/roxygen')`

File: R/new_variable.R [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#' Creates a new variable in the dataframe.
+#' @param df dataframe. The dataframe to modify.
+#' @param fun function. The function to run to convert inputs to outputs.
+#' @param inputs character. The names of the variables needed to make the output.
+#' @param output character. The name of the desired new variable.
+#' @export
+new_variable <- function(df, fun, output, inputs = NULL) {
+  if (is.null(inputs)) inputs <- as.character(names(formals(fun)))
```
Comment (robertzk) [inline R/new_variable.R]: You mean `inputs <<-`?

Comment (peterhurford) [inline R/new_variable.R]: Nope, not that kind of `inputs`.  I'll redefine to clarify.

**Comment by peterhurford** [comment]
@robertzk This has tests and documentation now.  Good to go?

**Pull request merged**